src/FileConventions/Library.fs

```diff
@@ -104,3 +104,63 @@ let EolAtEof(fileInfo: FileInfo) =
                 False
         else
             True
+
+let DetectInconsistentVersionsInGitHubCIWorkflow(fileInfos: seq<FileInfo>) =
+    fileInfos
+    |> Seq.iter(fun fileInfo -> assert (fileInfo.FullName.EndsWith ".yml"))
+
+    let inconsistentPulumiVersions =
+        fileInfos
+        |> Seq.map(fun fileInfo -> File.ReadLines fileInfo.FullName)
+        |> Seq.map(fun fileLines ->
+            fileLines
+            |> Seq.filter(fun line -> line.Contains "pulumi-version:")
+            |> Seq.map(fun line -> line.Substring(line.IndexOf(":") + 1))
+            |> Seq.map(fun line -> line.Trim())
+        )
+        |> Seq.concat
+        |> Set.ofSeq
+        |> Seq.length
+        |> (fun length -> length > 1)
+
+    let versionRegex = Regex("\\s([^\\s]*)@([^\\s]*)\\s", RegexOptions.Compiled)
+
+    let dict: Dictionary<string, Set<string>> =
+        new Dictionary<string, Set<string>>()
+
+    fileInfos
+    |> Seq.iter(fun fileInfo ->
+        let fileText = File.ReadAllText fileInfo.FullName
+
+        (versionRegex.Matches fileText)
+        |> Seq.iter(fun regexMatch ->
+            let key = regexMatch.Groups.[1].ToString()
+            let value = regexMatch.Groups.[2].ToString()
+
+            if dict.ContainsKey(key) then
+                dict.[key] <- Set.add value dict[key]
+            else
+                dict.[key] <- Set.singleton(value)
+        )
+    )
+
+    let inconsistentVersions =
+        dict
+        |> Seq.map(fun item -> Seq.length item.Value > 1)
+        |> Seq.contains true
+
+    inconsistentPulumiVersions || inconsistentVersions
+
+let DetectInconsistentVersionsInGitHubCI(directoryInfo: DirectoryInfo) =
```

Comment:
@realmarv I know the type is called DirectoryInfo, but you don't need to name the var in the same way; it can be much shorter, just `dir` is fine

Reply:
Done